### PR TITLE
Update to v24.1.0-IOFreeze

### LINKED ISSUE #5090: Update to EnergyPlus 24.1.0
Enhancement Request
--------------
Once v24.1-IOFreeze is released, we should update to it.

## Detailed Description
https://github.com/NREL/EnergyPlus/blob/develop/src/Transition/InputRulesFiles/Rules23-2-0-to-24-1-0.md
https://github.com/NREL/EnergyPlus/blob/develop/src/Transition/OutputRulesFiles/OutputChanges23-2-0-to-24-1-0.md

### Major changes
- [ ] `HeatExchanger:AirToAir:SensibleAndLatent`: fields removed, fields added
  - https://github.com/NREL/OpenStudio/pull/5099

### New additions
- [ ] `IndoorLivingWall`: new object
  - https://github.com/NREL/OpenStudio/issues/5095
- [ ] `ZoneHVAC:PackagedTerminalAirConditioner`, `ZoneHVAC:PackagedTerminalHeatPump`, `ZoneHVAC:WaterToAirHeatPump`, `AirLoopHVAC:UnitarySystem`: new "No Load Supply Air Flow Rate Control Set To Low Speed" field
  - https://github.com/NREL/OpenStudio/pull/5096

### Minor stuff
- [ ] `People`: field "Mean Radiant Temperature Calculation Type" key ZoneAveraged changed to EnclosureAveraged
  - https://github.com/NREL/OpenStudio/pull/5100
- [ ] `Lights`, `ElectricEquipment`, `GasEquipment`, `HotWaterEquipment`, `SteamEquipment`, `OtherEquipment`, `ElectricEquipment:ITE:AirCooled`: field "xxx per Zone Floor Area" renamed to "xxx per Floor Area"
  - **Why is `ElectricEquipment:ITE:AirCooled` still "Watts per Zone Floor Area" in OS idd?**
- [x] `ZoneHVAC:TerminalUnit:VariableRefrigerantFlow`: min fields changed
- [x] `Boiler:HotWater`: field "Nominal Thermal Efficiency" removes maximum
- [x] `WaterHeater:Mixed`, `WaterHeater:Stratified`: field "Heater Thermal Efficiency" removes maximum
- [x] `OutputControl:Table:Style`: field "Unit Conversion" has new key
- [x] `Output:SQLite`: field "Unit Conversion for Tabular Data" has new key

### Other
- [x] `ComfortViewFactorAngles`: remove "Zone Name" field
  - **Appears this isn't wrapped; no OS change required?**
- [x] `AirflowNetwork:Distribution:Component:Coil`: field "Coil Object Type" has new keys
  - **Appears this isn't wrapped; no OS change required?**
- [x] `ZoneHVAC:ForcedAir:UserDefined`, `AirTerminal:SingleDuct:UserDefined`, `Coil:UserDefined`, `PlantComponent:UserDefined`: extra notes

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Fixes #5090

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Review by jmarrec — APPROVED** [review]
Pretty straightforward for that portion.

File: resources/energyplus/ProposedEnergy+.idd [hunk 1]
```diff
@@ -311,6 +311,7 @@
 !      percent/K              =>   percent/F           0.555555555555556
 !      person/m2              =>   person/ft2          0.0928939733269818
 !      s/m                    =>   s/ft                0.3048
+!      umol/m2-s              =>   umol/ft2-s          0.09290304
```
Comment (joseph-robertson) [inline resources/energyplus/ProposedEnergy+.idd:314]: Thanks!

**Comment by jmarrec** [comment]
Tests are ok. merging.

**Comment by jmarrec** [comment]
Ideally we'll need to build E+ for Centos and update the hash, but I don't have the machine I use to build the centos package right now. It's not a big deal

**Pull request merged**